src/Faithlife.Tracing/AspNet/TracingInitializationInfo.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Faithlife.Tracing.AspNet
+{
+	public sealed class TracingInitializationInfo
+	{
+		public string ServiceName { get; set; }
+		public Func<ITraceFactory> CreateTraceFactory { get; set; }
```

Comment:
What's the advantage of the `Func<>`?

Reply:
So it can be lazy. `Initialize` is called multiple times, due to the way this integrates into the ASP.NET lifecycle.

Or we could split this into two initialization methods; more lines of code to write for clients but perhaps a cleaner API?

Reply:
I like clean APIs. Another advantage would be potentially selective application of `TracingActionFilterAttribute`?